mlir/lib/Transforms/lab4/vasilev_ivan/dephCounter.cpp

```diff
@@ -0,0 +1,59 @@
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Tools/Plugins/PassPlugin.h"
+
+#include "mlir/IR/BuiltinOps.h"
+
+using namespace mlir;
+
+namespace {
+
+class DepthCounter
+    : public PassWrapper<DepthCounter, OperationPass<func::FuncOp>> {
+public:
+  StringRef getArgument() const final { return "VasilevDepthCounter"; }
+  StringRef getDescription() const final {
+    return "Count the maximum depth of region nests in a function";
+  }
+
+  void runOnOperation() override {
+
+    getOperation()->walk([&](Operation *op) {
+      int maxDepth = computeMaxDepth(op);
+
+      op->setAttr(
+          "max_region_depth",
+          IntegerAttr::get(IntegerType::get(op->getContext(), 32), maxDepth));
+    });
+  }
+
+private:
+  int computeMaxDepth(Operation *op, int currentDepth = 0) {
+    int maxDepth = currentDepth;
+
+    for (Region &region : op->getRegions()) {
+      for (Block &block : region) {
+        for (Operation &nestedOp : block) {
+          maxDepth =
+              std::max(maxDepth, computeMaxDepth(&nestedOp, currentDepth + 1));
+        }
+      }
+    }

```

Comment:
It looks like you are doing to many operations here. You call the same function on each operation, so on you calculate the same value multiple times on each region